collect_app/src/main/java/org/odk/collect/android/widgets/QuestionWidget.java

```diff
@@ -443,13 +456,28 @@ protected ImageView getAnswerImageView(Bitmap bitmap) {
             @Override
             public void onClick(View v) {
                 if (questionWidget instanceof BaseImageWidget) {
-                    ((BaseImageWidget) questionWidget).onImageClick();
+                    if (imageView.isClickable()) {
+                        disableViewForOneSecond(imageView);
+                        ((BaseImageWidget) questionWidget).onImageClick();
+                    }
                 }
             }
         });
         return imageView;
     }
 
+    // This method is used to avoid opening more than one dialog or activity when user quickly clicks the button several times:
+    // https://github.com/opendatakit/collect/issues/1624
+    protected void disableViewForOneSecond(final View view) {
```

Comment:
Is this the best way to do this? There's no way to disable input until the Dialog pops up? I'm super wary of anything time based.

Reply:
It's not related to Dialogs only. it refers to activities too. I was thinking about RxBinding https://code.tutsplus.com/tutorials/rxjava-for-android-apps-introducing-rxbinding-and-rxlifecycle--cms-28565 there is an example:

RxView.clicks(button)
    .debounce(500, TimeUnit.MILLISECONDS)
    .subscribe(aVoid -> {
      Toast.makeText(MainActivity.this, "RxView.clicks", Toast.LENGTH_SHORT).show();
    });

But I think it would be more helpful when it comes to https://github.com/opendatakit/collect/issues/1624 and then I'll investigate it deeper.

Reply:
The RxBinding approach you're looking at there is basically the same code written a different way, right @grzesiek2010? In other words, it's still time-based. 

I think alternatives would be things like disabling the clickable thing until whatever needs to happen (dialog popup, etc) completes either with a global boolean or with code in callbacks.

Reply:
Yes RXBinding it's almost the same but it will be more convenient to use if we need to protect lots of buttons etc (in this case we protect all buttons but everything is in one place in QuestionWidget class).
That's why I said RXBinding might be a better solution for #1624.

I think your idea would be really difficult and complex and I'm not in favor of it.  

"until whatever needs to happen"
then we would have to remember about it and set such a global value in lots o different places. I think my solution is better since everything is in on onClick() method.

Reply:
I agree that the timing approach is simpler and less error-prone than anything else I'm seeing. @jknightco any other ideas to explore?

How about making the debounce time 500ms so it's less likely to block a user from real, intentional interaction.

Reply:
500ms should be enough too. But I need to check it and confirm.